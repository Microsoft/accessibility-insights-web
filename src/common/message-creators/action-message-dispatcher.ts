// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { PayloadWithEventName } from 'background/actions/action-payloads';
import { TelemetryData } from '../extension-telemetry-events';
import { InterpreterMessage, Message } from '../message';
import { Messages } from '../messages';
import { Dispatcher } from './types/dispatcher';

export class ActionMessageDispatcher implements Dispatcher {
    constructor(private postMessageDelegate: (message: InterpreterMessage) => void, private tabId: number) {}

    public dispatchMessage(message: Message): void {
        const interpreterMessage = this.decorateWithTabId(message);

        this.postMessageDelegate(interpreterMessage);
    }

    public dispatchType(messageType: string): void {
        this.dispatchMessage({
            messageType,
        });
    }

    public sendTelemetry(eventName: string, eventData: TelemetryData): void {
        const payload: PayloadWithEventName = {
            eventName: eventName,
            telemetry: eventData,
        };

        const message: Message = {
            messageType: Messages.Telemetry.Send,
            payload,
        };

        this.dispatchMessage(message);
    }

    private decorateWithTabId(message: Message): InterpreterMessage {
        const decorated: InterpreterMessage = {
            ...message,
        };

        if (this.tabId != null) {
            decorated.tabId = this.tabId;
        }

        return decorated;
    }
}
