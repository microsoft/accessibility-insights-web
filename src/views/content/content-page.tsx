// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { HyperlinkDefinition } from 'common/types/hyperlink-definition';
import { flatten, forEach, toPairs } from 'lodash';
import * as React from 'react';

import { MarkupBasedComponentProps, createMarkup, Markup, MarkupDeps } from './markup';

type HyperlinkDefinitionMap = { [KEY in string]: { href: string; text: string } };
type HyperlinkComponentMap<M extends HyperlinkDefinitionMap> = {
    [KEY in keyof M]: React.FC<React.PropsWithChildren<unknown>>;
};

export type ContentPageDeps = MarkupDeps;
export type ContentPageProps = MarkupBasedComponentProps;

export type ContentPageComponent = React.FC<React.PropsWithChildren<ContentPageProps>> & {
    displayName: 'ContentPageComponent';
    pageTitle?: string;
};

export type ContentReference = string | ContentPageComponent;
type CreateProps<M extends HyperlinkDefinitionMap> = {
    Markup: Markup;
    Link: HyperlinkComponentMap<M>;
};
export function ContentCreator<M extends HyperlinkDefinitionMap>(
    linkMap?: M,
): (fn: (props: CreateProps<M>) => JSX.Element) => ContentPageComponent {
    function mapLinks(markup: Markup): HyperlinkComponentMap<M> {
        const map: Partial<HyperlinkComponentMap<M>> = {};

        forEach(linkMap, (hyperlink: HyperlinkDefinition, key: keyof M) => {
            map[key] = ({ children }) => (
                <markup.HyperLink href={hyperlink.href}>
                    {children || hyperlink.text}
                </markup.HyperLink>
            );
        });

        return map as HyperlinkComponentMap<M>;
    }

    function create(fn: (props: CreateProps<M>) => JSX.Element): ContentPageComponent {
        return NamedFC<ContentPageProps>('ContentPageComponent', props => {
            const { deps, options } = props;
            const markup = createMarkup(deps, options);
            return fn({ Markup: markup, Link: mapLinks(markup) });
        }) as ContentPageComponent;
    }

    return create;
}

export interface ContentProvider {
    getPage(path: string): ContentPageComponent;
    allPaths(): string[];
    pathTo(component: ContentPageComponent): string | null;
    contentFromReference(content?: ContentReference): ContentPageComponent;
    pathFromReference(content?: ContentReference): string | null;
}
type ContentTree = { [K in string]: ContentTree | ContentPageComponent };
export function ContentProvider(root: ContentTree): ContentProvider {
    const create = ContentCreator();

    const notFoundPage = (path?: string) => create(() => <h1>Cannot find {path}</h1>);

    function isContentPageComponent(
        leaf: any | ContentPageComponent,
    ): leaf is ContentPageComponent {
        return (
            leaf &&
            (leaf as ContentPageComponent).displayName &&
            (leaf as ContentPageComponent).displayName === 'ContentPageComponent'
        );
    }

    type TreeEntry = { path: string; leaf: ContentPageComponent };
    const flattenTree = (tree: ContentTree): TreeEntry[] => {
        const prefixEntry =
            (prefix: string) =>
            ({ path, leaf }: TreeEntry) =>
                ({
                    path: prefix + '/' + path,
                    leaf,
                }) as TreeEntry;

        const entries = toPairs(tree).map(([key, leaf]) =>
            isContentPageComponent(leaf)
                ? ({ path: key, leaf } as TreeEntry)
                : flattenTree(leaf).map(prefixEntry(key)),
        );

        return flatten(entries);
    };
    const rootEntries = flattenTree(root);

    function findPage(tree: ContentTree, [head, ...tail]: string[]): ContentPageComponent | null {
        if (!tree) {
            return null;
        }

        const branch = tree[head];
        if (isContentPageComponent(branch)) {
            if (tail.length === 0) {
                return branch;
            } else {
                return null;
            }
        } else {
            return findPage(branch, tail);
        }
    }

    function getPage(path?: string): ContentPageComponent {
        return (path && findPage(root, path.split('/'))) || notFoundPage(path);
    }

    const allPaths = () => rootEntries.map(entry => entry.path);

    const pathTo = (component: ContentPageComponent) => {
        const entry = rootEntries.find(e => e.leaf === component);
        return entry ? entry.path : null;
    };

    const contentFromReference = (reference?: ContentReference) =>
        isContentPageComponent(reference) ? reference : getPage(reference);
    const pathFromReference = (reference?: ContentReference) =>
        isContentPageComponent(reference) ? pathTo(reference) : reference ?? null;

    return { getPage, allPaths, pathTo, contentFromReference, pathFromReference };
}

export const ContentPage = {
    create: ContentCreator(),
    provider: ContentProvider,
};
