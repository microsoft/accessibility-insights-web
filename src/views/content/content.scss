// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/icons/icon.scss';
@import '../../common/styles/colors.scss';
@import '../../common/styles/fonts.scss';

.content-container {
    display: flex;

    .content-left,
    .content-right {
        flex-grow: 1;
        max-width: 150px;
    }
}

.content-header {
    @include text-style-title-l;

    margin-left: 10px;
    margin-right: 10px;
}

.content {
    flex-grow: 10;
    max-width: 1120px;
    font-family: $font-family;
    font-size: 14px;
    margin: 10px;
    word-break: break-word;

    table {
        font-family: $font-family;
        font-size: 14px;
    }

    h1 {
        @include text-style-title-l;
    }

    h2 {
        font-size: 21px;
        font-weight: 700;
        letter-spacing: -0.02em;
        line-height: 32px;
    }

    h3 {
        font-size: 14px;
    }

    h4 {
        font-weight: 400;
        font-size: 14px;
        margin-bottom: 4px;
    }

    h5 {
        font-weight: 700;
        font-size: 14px;
        margin-bottom: 4px;
    }

    a {
        font-weight: 400;
    }

    ol {
        -webkit-padding-start: 16px; // Chrome versions before 87 use non-standard name: https://caniuse.com/mdn-css_properties_padding-inline-start
        padding-inline-start: 16px;

        li ul li {
            list-style-type: disc;
        }
    }

    .content-hyperlinks {
        .insights-link {
            line-height: 20px;
            margin-bottom: 6px;
            display: block;
        }

        .content-inline {
            line-height: 20px;
            margin-bottom: 6px;

            .insights-link {
                display: inline;
            }
        }
    }

    .code-example {
        .code-example-code {
            font-family: $code-font-family;
            margin-block-start: 1em;
            margin-block-end: 1em;
        }

        .code-example-title {
            h4 {
                display: inline;
                margin-block-start: 0;
            }
        }
    }

    .columns {
        margin-top: 40px;
        @media screen and (width >= 800px) {
            display: flex;
        }
    }

    .column {
        padding: 10px;
        flex-grow: 1;
        flex-basis: 0;
        max-width: 520px;
    }

    .column + .column {
        @media screen and (width >= 800px) {
            margin-left: 80px;
        }
    }

    $icon-size: 16px;
    $do-color: $positive-outcome;
    $dont-color: $negative-outcome;

    .insights-code {
        font-family: $code-font-family;
    }

    .pass-fail-grid {
        display: grid;
        grid-template-areas:
            'fail-section pass-section'
            'fail-example pass-example';
        border: 1px solid $neutral-60;
        word-break: break-word;
    }

    .fail-section,
    .fail-example,
    .pass-section,
    .pass-example {
        padding: 15px;
    }

    .fail-example,
    .pass-example {
        border-top: 1px solid $neutral-60;
    }

    .fail-section,
    .fail-example {
        border-right: 1px solid $neutral-60;
    }

    .fail-section {
        grid-area: fail-section;
    }

    .fail-example {
        grid-area: fail-example;
    }

    .pass-section {
        grid-area: pass-section;
    }

    .pass-example {
        grid-area: pass-example;
    }

    .do-header,
    .pass-header,
    .dont-header,
    .fail-header {
        width: 100%;
    }

    .do-header,
    .dont-header {
        .check-container {
            margin-top: 9px;
            float: right;
        }

        h2 {
            display: inline;
        }
    }

    .pass-header,
    .fail-header {
        h3 {
            display: inline;
        }
    }

    .do-section,
    .pass-section,
    .pass-example {
        .highlight {
            color: $do-color;
            font-weight: bold;
        }
    }

    .dont-section,
    .fail-section,
    .fail-example {
        .highlight {
            color: $dont-color;
            font-weight: bold;
        }
    }

    .do-header {
        @include check-icon-styles($icon-size, 0, $do-color);

        border-bottom: 2px solid $do-color;
    }

    .dont-header {
        @include cross-icon-styles($icon-size, 0, $dont-color);

        border-bottom: 2px solid $dont-color;
    }

    .pass-header {
        @include check-icon-styles(12px, 0, $do-color);
    }

    .fail-header {
        @include cross-icon-styles(12px, 0, $dont-color);
    }

    .table {
        list-style: none;
        margin-top: 12px;
        display: table;
        border-collapse: separate;
        border-spacing: 5px;
        padding-left: 18px;
    }

    .accessibility-problems-list {
        background: $neutral-6;
        list-style-type: disc;
    }

    .landmark-table {
        border-spacing: 5px;

        @media screen and (width >= 600px) {
            padding-left: 32px;
        }

        .landmark-table-column {
            width: 24%;
        }

        .landmark-role {
            text-align: center;
            border-style: dashed;
            border-width: 3px;
        }

        td,
        th {
            padding: 5px;
            vertical-align: text-top;
        }

        th {
            background: $neutral-100;
            color: $neutral-0;
            font-weight: 400;
        }

        td {
            background: $neutral-10;
            color: $neutral-100;
        }

        .contentinfo-landmark {
            border-color: $landmark-contentinfo;
        }

        .main-landmark {
            border-color: $landmark-main;
        }

        .complementary-landmark {
            border-color: $landmark-complementary;
        }

        .banner-landmark {
            border-color: $landmark-banner;
        }

        .region-landmark {
            border-color: $landmark-region;
        }

        .navigation-landmark {
            border-color: $landmark-navigation;
        }

        .search-landmark {
            border-color: $landmark-search;
        }

        .form-landmark {
            border-color: $landmark-form;
        }
    }

    .landmarks-legend {
        color: $neutral-100;
        font-size: 14px;
        display: table-cell;
        width: 110px;
        text-align: center;
        font-weight: 400;
        padding: 4px;
        vertical-align: middle;
        border-style: dashed;
        border-width: 3px;

        &.contentinfo-landmark {
            border-color: $landmark-contentinfo;
        }

        &.main-landmark {
            border-color: $landmark-main;
        }

        &.complementary-landmark {
            border-color: $landmark-complementary;
        }

        &.banner-landmark {
            border-color: $landmark-banner;
        }

        &.region-landmark {
            border-color: $landmark-region;
        }

        &.navigation-landmark {
            border-color: $landmark-navigation;
        }

        &.search-landmark {
            border-color: $landmark-search;
        }

        &.form-landmark {
            border-color: $landmark-form;
        }
    }
}
