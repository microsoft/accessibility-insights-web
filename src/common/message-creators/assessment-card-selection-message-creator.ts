// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    AssessmentCardSelectionPayload,
    AssessmentCardToggleVisualHelperPayload,
    AssessmentExpandCollapsePayload,
    AssessmentSingleRuleExpandCollapsePayload,
} from 'background/actions/action-payloads';
import { TelemetryEventSource } from 'common/extension-telemetry-events';
import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
import { ActionMessageDispatcher } from 'common/message-creators/types/dispatcher';
import { AssessmentCardSelectionMessages } from 'common/messages';
import { SupportedMouseEvent, TelemetryDataFactory } from 'common/telemetry-data-factory';

export class AssessmentCardSelectionMessageCreator implements CardSelectionMessageCreator {
    constructor(
        private readonly dispatcher: ActionMessageDispatcher,
        private readonly telemetryFactory: TelemetryDataFactory,
        private readonly source: TelemetryEventSource,
        protected readonly messages: AssessmentCardSelectionMessages,
    ) {}

    public toggleCardSelection = (
        ruleId: string,
        resultInstanceUid: string,
        event: React.SyntheticEvent,
        testKey: string,
    ) => {
        const payload: AssessmentCardSelectionPayload = {
            testKey,
            resultInstanceUid,
            ruleId,
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: this.messages.CardSelectionToggled,
            payload,
        });
    };

    public toggleRuleExpandCollapse = (
        ruleId: string,
        event: React.SyntheticEvent,
        testKey: string,
    ) => {
        const payload: AssessmentSingleRuleExpandCollapsePayload = {
            testKey,
            ruleId,
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: this.messages.RuleExpansionToggled,
            payload,
        });
    };

    public collapseAllRules = (event: SupportedMouseEvent, testKey: string) => {
        const payload: AssessmentExpandCollapsePayload = {
            testKey,
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: this.messages.CollapseAllRules,
            payload,
        });
    };

    public expandAllRules = (event: SupportedMouseEvent, testKey: string) => {
        const payload: AssessmentExpandCollapsePayload = {
            testKey,
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: this.messages.ExpandAllRules,
            payload,
        });
    };

    public toggleVisualHelper = (event: SupportedMouseEvent, testKey: string) => {
        const payload: AssessmentCardToggleVisualHelperPayload = {
            testKey,
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: this.messages.ToggleVisualHelper,
            payload,
        });
    };
}
