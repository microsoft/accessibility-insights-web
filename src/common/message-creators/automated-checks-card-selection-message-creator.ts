// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    BaseActionPayload,
    CardSelectionPayload,
    RuleExpandCollapsePayload,
} from 'background/actions/action-payloads';
import { TelemetryEventSource } from 'common/extension-telemetry-events';
import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
import { ActionMessageDispatcher } from 'common/message-creators/types/dispatcher';
import { Messages } from 'common/messages';
import { SupportedMouseEvent, TelemetryDataFactory } from 'common/telemetry-data-factory';

export class AutomatedChecksCardSelectionMessageCreator implements CardSelectionMessageCreator {
    constructor(
        private readonly dispatcher: ActionMessageDispatcher,
        private readonly telemetryFactory: TelemetryDataFactory,
        private readonly source: TelemetryEventSource,
    ) {}

    public toggleCardSelection = (
        ruleId: string,
        resultInstanceUid: string,
        event: React.SyntheticEvent,
    ) => {
        const payload: CardSelectionPayload = {
            resultInstanceUid,
            ruleId,
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: Messages.CardSelection.CardSelectionToggled,
            payload,
        });
    };

    public toggleRuleExpandCollapse = (ruleId: string, event: React.SyntheticEvent) => {
        const payload: RuleExpandCollapsePayload = {
            ruleId,
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: Messages.CardSelection.RuleExpansionToggled,
            payload,
        });
    };

    public collapseAllRules = (event: SupportedMouseEvent) => {
        const payload: BaseActionPayload = {
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: Messages.CardSelection.CollapseAllRules,
            payload,
        });
    };

    public expandAllRules = (event: SupportedMouseEvent) => {
        const payload: BaseActionPayload = {
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: Messages.CardSelection.ExpandAllRules,
            payload,
        });
    };

    public toggleVisualHelper = (event: SupportedMouseEvent) => {
        const payload: BaseActionPayload = {
            telemetry: this.telemetryFactory.withTriggeredByAndSource(event, this.source),
        };

        this.dispatcher.dispatchMessage({
            messageType: Messages.CardSelection.ToggleVisualHelper,
            payload,
        });
    };
}
