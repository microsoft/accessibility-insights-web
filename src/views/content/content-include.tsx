// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedFC } from '../../common/react/named-fc';
import { ContentPageDeps, ContentProvider, ContentReference } from './content-page';

export type ContentIncludeDeps = {
    contentProvider: ContentProvider;
} & ContentPageDeps;

export type ContentIncludeProps = {
    deps: ContentIncludeDeps;
    content: ContentReference | undefined;
};

export const ContentInclude = NamedFC<ContentIncludeProps>(
    'ContentInclude',
    ({ deps, content }) => {
        const { contentProvider } = deps;

        if (!content) {
            return null;
        }

        const Content = contentProvider.contentFromReference(content);

        return (
            <div className="content">
                <Content deps={deps} />
            </div>
        );
    },
);
