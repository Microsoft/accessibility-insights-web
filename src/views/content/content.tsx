// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../common/react/named-sfc';
import { ContentPageDeps, ContentProvider, ContentReference } from './content-page';
import { Page } from '../page/page';

export type ContentDeps = { contentProvider: ContentProvider } & ContentPageDeps;

export type ContentProps = { deps: ContentDeps; reference: ContentReference };

export const Content = NamedSFC<ContentProps>('Content', ({ deps, reference }) => {
    const { contentProvider } = deps;
    const ContentPage = contentProvider.contentFromReference(reference);
    return (
        <Page>
            <div className="content-container">
                <div className="content-left" />
                <div className="content">
                    <ContentPage deps={deps} options={{ setPageTitle: true }} />
                </div>
                <div className="content-right" />
            </div>
        </Page>
    );
});
