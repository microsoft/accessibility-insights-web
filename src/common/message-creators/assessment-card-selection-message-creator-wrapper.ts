// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { BaseStore } from 'common/base-store';
import { VisualizationConfigurationFactory } from 'common/configs/visualization-configuration-factory';
import { AssessmentCardSelectionMessageCreator } from 'common/message-creators/assessment-card-selection-message-creator';
import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
import { SupportedMouseEvent } from 'common/telemetry-data-factory';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';

export const NO_ASSESSMENT_STORE_DATA_ERROR = 'no assessment store data';

export class AssessmentCardSelectionMessageCreatorWrapper implements CardSelectionMessageCreator {
    constructor(
        private assessmentCardSelectionMessageCreator: AssessmentCardSelectionMessageCreator,
        private visualizationFactory: VisualizationConfigurationFactory,
        private assessmentStore: BaseStore<AssessmentStoreData, Promise<void>>,
    ) {}

    public toggleVisualHelper = (event: SupportedMouseEvent) => {
        const testKey = this.getSelectedTestKey();
        return this.assessmentCardSelectionMessageCreator.toggleVisualHelper(event, testKey);
    };

    public expandAllRules = (event: SupportedMouseEvent) => {
        const testKey = this.getSelectedTestKey();
        return this.assessmentCardSelectionMessageCreator.expandAllRules(event, testKey);
    };

    public collapseAllRules = (event: SupportedMouseEvent) => {
        const testKey = this.getSelectedTestKey();
        return this.assessmentCardSelectionMessageCreator.collapseAllRules(event, testKey);
    };

    public toggleCardSelection = (
        ruleId: string,
        resultInstanceUid: string,
        event: React.SyntheticEvent,
    ) => {
        const testKey = this.getSelectedTestKey();
        return this.assessmentCardSelectionMessageCreator.toggleCardSelection(
            ruleId,
            resultInstanceUid,
            event,
            testKey,
        );
    };

    public toggleRuleExpandCollapse = (ruleId: string, event: React.SyntheticEvent) => {
        const testKey = this.getSelectedTestKey();
        return this.assessmentCardSelectionMessageCreator.toggleRuleExpandCollapse(
            ruleId,
            event,
            testKey,
        );
    };

    private getSelectedTestKey = () => {
        const assessmentStoreState = this.assessmentStore.getState();

        if (assessmentStoreState == null) {
            throw NO_ASSESSMENT_STORE_DATA_ERROR;
        }

        const selectedTest = assessmentStoreState.assessmentNavState.selectedTestType;
        const testKey = this.visualizationFactory.getConfiguration(selectedTest).key;
        return testKey;
    };
}
