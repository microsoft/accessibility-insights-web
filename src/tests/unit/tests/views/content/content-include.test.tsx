// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import * as React from 'react';

import { ContentInclude } from 'views/content/content-include';
import { ContentPage } from 'views/content/content-page';
import { ContentActionMessageCreator } from '../../../../../common/message-creators/content-action-message-creator';

describe('ContentInclude', () => {
    const content = {
        for: {
            testing: ContentPage.create(() => 'CONTENT FOR TESTING' as any),
        },
    };

    const applicationTitle = 'THE_APPLICATION_TITLE';
    const deps = {
        textContent: {
            applicationTitle,
        },
        contentProvider: ContentPage.provider(content),
        contentActionMessageCreator: {} as any as ContentActionMessageCreator,
    };

    it('renders from content', () => {
        const renderResult = render(<ContentInclude deps={deps} content={content.for.testing} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it('renders from path', () => {
        const renderResult = render(<ContentInclude deps={deps} content={'for/testing'} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
