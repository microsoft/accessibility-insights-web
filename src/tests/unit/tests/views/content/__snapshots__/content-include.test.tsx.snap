// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentInclude renders from content 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    CONTENT FOR TESTING
  </div>
</DocumentFragment>
`;

exports[`ContentInclude renders from path 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    CONTENT FOR TESTING
  </div>
</DocumentFragment>
`;
